File: tests/primitives/str_literal_escape/str_literal_escape.fs
module rec str_literal_escape.Program

let f (_: char) = ()
let g (_: string) = ()

let main _ =
  f '\''
  f '\\'
  f '\x00'
  f '\x7f'
  f '\xFF'
  g "\"HELLO\n WORLD\""
  g "\x7f"

  // FIXME: escape invalid UTF-8 sequence
  g "\x80 \xc1"
  0

File: tests/primitives/str_literal_escape/str_literal_escape.c
#include <milone.h>

void str_literal_escape_Program_f(char arg_);

void str_literal_escape_Program_g(struct String arg_1);

int main(int argc, char** argv);

void str_literal_escape_Program_f(char arg_) {
    return;
}

void str_literal_escape_Program_g(struct String arg_1) {
    return;
}

int main(int argc, char** argv) {
    milone_start(argc, argv);
    str_literal_escape_Program_f('\'');
    str_literal_escape_Program_f('\\');
    str_literal_escape_Program_f('\0');
    str_literal_escape_Program_f('\x7f');
    str_literal_escape_Program_f((char)'\xff');
    str_literal_escape_Program_g((struct String){.str = "\"HELLO\n WORLD\"", .len = 14});
    str_literal_escape_Program_g((struct String){.str = "\x7f", .len = 1});
    str_literal_escape_Program_g((struct String){.str = "€ Á", .len = 3});
    return 0;
}
